folly/Traits.h

```diff
@@ -312,6 +312,36 @@ struct is_negative_impl<T, false> {
   constexpr static bool check(T) { return false; }
 };
 
+// Early versions of MSVC choke on the template
+// constraints.
+#if _MSC_FULL_VER <= 200022816 // 2015 RC or below
+
+FOLLY_PUSH_WARNING
```

Comment:
Ah, woops, hadn't realized that slipped into this PR. It's from #263, which hasn't been merged yet.